src/main/xerus/monstercat/tabs/TabPlaylist.kt

```diff
@@ -0,0 +1,80 @@
+package xerus.monstercat.tabs
+
+import javafx.scene.control.*
+import javafx.scene.input.KeyCode
+import javafx.scene.input.MouseButton
+import xerus.ktutil.javafx.MenuItem
+import xerus.ktutil.javafx.TableColumn
+import xerus.ktutil.javafx.fill
+import xerus.ktutil.javafx.properties.listen
+import xerus.monstercat.api.Player
+import xerus.monstercat.api.Playlist
+import xerus.monstercat.api.response.Track
+
+
+class TabPlaylist : VTab() {
+	private var table = TableView<Track>().apply {
+		// Inherent properties
+		columnResizePolicy = TableView.CONSTRAINED_RESIZE_POLICY
+		selectionModel.selectionMode = SelectionMode.SINGLE
+		placeholder = Label("Your playlist is empty.")
+
+		// Events
+		fun removeFromPlaylist() = Playlist.removeAt(selectedIndex)
+		fun playFromPlaylist() = Player.playFromPlaylist(selectedIndex)
+		fun playNextPlaylist() = Playlist.addNext(selectedTrack)
+
+		setOnMouseClicked { me ->
+			if (me.button == MouseButton.PRIMARY && me.clickCount == 2) {
```

Comment:
`when` can be used here as well

Reply:
d84109a9c4fdf432f88e85cb1c3e7fe0e6d8edf8